CeeFM/Decideify
# Movies.js and TVShows.js algorithm cleanup and rollout part 2

**Description**
YEAHHHH SUGGESTION ALGORITHMS ARE WORKING FOR MOVIES AND TV SHOWS!!!!

still need to iron out the details about what to do when suggestions come back with null descriptions and/or image pages, and also consider adding in the additional backdrop_path image for more stylizing? Make that decision later though once we have the other MVP features built out, hopefully soon!

**Type of Change**


**Testing Instructions**

- Clone this project’s repository to your machine
  -If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  -Then `git checkout main` to switch to the project’s main branch)
- Navigate to the root directory for this repo (yourParentFolderName/Decideify)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `cm-suggestions` as one of the available branches
- `git checkout cm-suggestions` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin cm-suggestions`
- `Start Decideify.sln` to open the project in Visual Studio
- If you haven’t created the Decideify database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/CeeFM/Decideify/blob/main/SQL/Decideify_Seed_Users_DB.sql)

- Click the execute with debugger solid green button to launch the app
- On the movie.js and tvshows.js pages:
  - You can choose a genre / type for the piece of content from a dropdown menu on each content hub page, and then test that submission and see two things happen
    - 1) the "CategoryId" property of the "suggestion" state variable should be updated to the Id of the option from the "Category" database table for the option you chose
    - 2) the entire Category object that you selected should also be printed to the console after being saved to another state variable named userCategory  
  - After you submit the genre / type form, the form should disappear and be replaced with an image that says YOUR NEW FAVORITE MOVIE or YOUR NEW FAVORITE TV SHOW depending on which hub you're on, as well as buttons to show a suggestion, and save that suggestion to your database. The save button's visibility at this point will be adjusted in a near-future pull request, as having the option to save something before you see it doesn't really make sense.
  - Clicking the Show Me button should print a random book to the console - if you selected a genre from the dropdown menu on the first page, the random book will be of that type / genre. if you didn't select a genre, it will return a random book of any type/genre.
  - You can also save any suggestion to your saved suggestions! This won't work in cases where expected information is null or unavailable, however we will handle that in a future pull request, intermittent failures to save suggestions are somewhat expected for now!

Checklist
